### PR TITLE
Support for SH1107

### LINKED ISSUE #341: support for ssd1107/sh1107
I recently bought a new OLED display on Aliexpress: https://www.aliexpress.com/item/4000547865501.html
The antistatic bag was labelled SH1107. The display has a resolution of 64x128 (it is vertical).
Datasheet is here: https://www.displayfuture.com/Display/datasheet/controller/SH1107.pdf
I tried to use the luma.oled library for control, but failed.
Can you please add support for this display as well? I can test. I tried to make some modifications in the library myself, but to no avail. For example, I tried to add a resolution of 64x128 to the SH1106 class (because some image, but shifted, could be achieved using this class), but failed. I didn't understand what the multiplex and displayoffset parameters were for.

Thanks.

### PR DESCRIPTION
**Pull request by thijstriemstra**
fixes #341

### CODE REVIEW

**Comment by thijstriemstra** [comment]
@rm-hull could you assist with the tests? I'm not sure how these calls are verified..

**Comment by rm-hull** [comment]
I just pushed b17e0731c4cb39563c9994ff2530f76b62ec0d41 - hopefully this should fix it. Those tests are useful for catching regressions, but are a bit of a PITA to write in the first place

**Review by rm-hull — APPROVED** [review]

File: luma/oled/device/__init__.py [hunk 1]
```diff
@@ -129,6 +133,87 @@ def display(self, image):
...
+    Serial interface to a monochrome SH1107 OLED display.
+
+    On creation, an initialization sequence is pumped to the display
+    to properly configure it. Further control commands can then be called to
+    affect the brightness and other settings.
+
+    :param serial_interface: The serial interface (usually a
+        :py:class:`luma.core.interface.serial.i2c` instance) to delegate sending
+        data and commands through.
+    :param width: The number of horizontal pixels (optional, defaults to 128).
```
Comment (thijstriemstra) [inline luma/oled/device/__init__.py:147]: should be 64

File: luma/oled/device/__init__.py [hunk 2]
```diff
@@ -129,6 +133,87 @@ def display(self, image):
...
+    :type width: int
+    :param height: The number of vertical pixels (optional, defaults to 64).
```
Comment (thijstriemstra) [inline luma/oled/device/__init__.py:149]: should be 128

**Pull request merged**